### PR TITLE
Add mini k8s deployment

### PR DESCRIPTION
**Pull request by MonkeyCanCode**
# Description

Add a mini k8s deployment for running polaris.

Fixes # [[DOC] Directions to run mini-deployment reference k8 directory that does not exist
](https://github.com/polaris-catalog/polaris/issues/99)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Here are all of the resources that got deployed with this deployment:
```
kubectl get all -n polaris
NAME                                      READY   STATUS    RESTARTS   AGE
pod/polaris-deployment-5b8457d65d-dhp5r   1/1     Running   0          108s

NAME                      TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
service/polaris-service   ClusterIP   10.96.173.228   <none>        8181/TCP   108s

NAME                                 READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/polaris-deployment   1/1     1            1           108s

NAME                                            DESIRED   CURRENT   READY   AGE
replicaset.apps/polaris-deployment-5b8457d65d   1         1         1       108s
```

On one session, performs a port forwarding for the svc:
```
kubectl port-forward svc/polaris-service -n polaris 8181:8181
Forwarding from 127.0.0.1:8181 -> 8181
Forwarding from [::1]:8181 -> 8181
```

On a new session, get the auth token from pod (in-memory deployment) then use polaris cli to perform auth:
```
kubectl  logs pods/polaris-deployment-5b8457d65d-dhp5r -n polaris | grep 'realm: default-realm root principal credentials:'
realm: default-realm root principal credentials: 2d54798a730ef589:b5b6ac216c5441e7421f241673837751
...
./polaris --client-id 2d54798a730ef589 --client-secret b5b6ac216c5441e7421f241673837751 principals list
{"name": "root", "clientId": "2d54798a730ef589", "properties": {}, "createTimestamp": 1723000008560, "lastUpdateTimestamp": 1723000008560, "entityVersion": 1}
```

# Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] If adding new functionality, I have discussed my implementation with the community using the linked GitHub issue
- [x] I have signed and submitted the [ICLA](https://github.com/polaris-catalog/polaris/blob/main/ICLA.md) and if needed, the [CCLA](https://github.com/polaris-catalog/polaris/blob/main/CCLA.md). See [Contributing](https://github.com/polaris-catalog/polaris/blob/main/CONTRIBUTING.md) for details.

### CODE REVIEW

**Comment by flyrain** [comment]
Thanks @MonkeyCanCode for working on this. We will need updates on the doc(README.md) as well. This part isn't consistent with the behavior:
Screenshot 2024-08-06 at 6 13 28 PM

We will need to add `-n polaris` for these commands, as well as to port forwarding command. I feel like these instructions are not necessarily in README, they can be in a separated doc, which is linked by the README, as we want to keep README lean.

**Comment by MonkeyCanCode** [comment]
> Thanks @MonkeyCanCode for working on this. We will need updates on the doc(README.md) as well. This part isn't consistent with the behavior:
> Screenshot 2024-08-06 at 6 13 28 PM
> 
> We will need to add `-n polaris` for these commands, as well as to port forwarding command. I feel like these instructions are not necessarily in README, they can be in a separated doc, which is linked by the README, as we want to keep README lean. 
> 

Got it. Will update it tonight

**Comment by MonkeyCanCode** [comment]
> > Thanks @MonkeyCanCode for working on this. We will need updates on the doc(README.md) as well. This part isn't consistent with the behavior:
> > Screenshot 2024-08-06 at 6 13 28 PM
> > We will need to add `-n polaris` for these commands, as well as to port forwarding command. I feel like these instructions are not necessarily in README, they can be in a separated doc, which is linked by the README, as we want to keep README lean.
> 
> Got it. Will update it tonight

This is completed and please take another look.

File: README.md [hunk 1]
```diff
@@ -57,9 +57,10 @@ Running in Docker
 
 Running in Kubernetes
 - `./setup.sh` - To run Polaris as a mini-deployment locally. This will create two pods that bind themselves to port `8181`.
-- `kubectl get pods` - To check the status of the pods.
-- `kubectl get deployment` - To check the status of the deployment.
-- `kubectl describe deployment polaris-deployment` - To troubleshoot if things aren't working as expected.
+- `kubectl get pods -n polaris` - To check the status of the pods.
+- `kubectl get deployment -n polaris` - To check the status of the deployment.
+- `kubectl describe deployment polaris-deployment -n polaris` - To troubleshoot if things aren't working as expected.
+- `kubectl port-forward svc/polaris-service -n polaris 8181:8181` - To create a secure connection between a local machine and a pod within the cluster.
```
Comment (flyrain) [inline README.md:63]: I think we should move this line right after line 59, as it is a part of k8s local deployment, or even better, if we can move it inside setup.sh?

**Review by flyrain — APPROVED** [review]
LGTM

**Comment by flyrain** [comment]
Thanks @MonkeyCanCode for the PR.

**Pull request merged**